### ISSUE TITLE
Could not get unknown property 'bundleReleaseAar' for object

### ISSUE THREAD

**Issue by NavalKishor** [sky-uk/gradle-maven-plugin#24]
project.ext {
    mavDevelopers = ["en":"Naval"]
    mavSiteUrl = "giturl"
    mavGitUrl = mavSiteUrl + '.git'
    mavProjectName = 'Communication'
    mavPublishToInternalRepo = true
    mavRepoInternalUrl =  "${System.env.HOME}/.m2/repository/"
    mavLibraryLicenses = ["Apache-2.0":'http://www.apache.org/licenses/LICENSE-2.0.txt']
    mavLibraryDescription = "This Libray help in communicating with the server"
    mavPluginVersion = '1.0.3'  
}

apply from: "https://raw.githubusercontent.com/sky-uk/gradle-maven-plugin/${project.mavPluginVersion}/gradle-mavenizer.gradle"

FAILURE: Build failed with an exception.

* Where:
Script 'https://raw.githubusercontent.com/sky-uk/gradle-maven-plugin/1.0.3/gradle-mavenizer.gradle' line: 13

* What went wrong:
A problem occurred configuring project ':myLibModule'.
> Could not get unknown property 'bundleReleaseAar' for object of type org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication.

**Comment by Fed93** [comment]
@NavalKishor Can you please post your build.gradle file?

**Comment by NavalKishor** [comment]
buildscript {
    repositories {
        mavenLocal()
        mavenCentral()
    }
}
plugins {
    // This plugin allows publishing android libraries
    id "com.github.dcendents.android-maven" version "2.1"
}
plugins {
    // This plugin allows publishing android libraries
    id "com.jfrog.bintray" version "1.8.4"

}
apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-kapt'
apply plugin: 'kotlin-android-extensions'

apply plugin: 'com.github.dcendents.android-maven'
apply plugin: 'com.jfrog.bintray'
apply plugin: 'maven-publish'
android {
    compileSdkVersion 28
    defaultConfig {
        minSdkVersion 21
        targetSdkVersion 28
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        multiDexEnabled true
    }
    flavorDimensions "default"
     productFlavors {
     Test{
	//…
	}
     INT{
	//….
	}
    Prod{
	//….
	}
    }
    buildTypes {
        debug {
            testCoverageEnabled true
            minifyEnabled false
            debuggable true
            jniDebuggable true
            renderscriptDebuggable true
            zipAlignEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
            }
     }
    lintOptions {  abortOnError false }
   testOptions { unitTests.returnDefaultValues = true} 
}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    api 'androidx.lifecycle:lifecycle-livedata-ktx:2.1.0'
}


task sourcesJar(type: Jar) {
    classifier = 'sources'
    from android.sourceSets.main.java.srcDirs
}

task javadoc(type: Javadoc) {
    source = android.sourceSets.main.java.srcDirs
    classpath += configurations.compile
    classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
    failOnError = false
}

task javadocJar(type: Jar, dependsOn: javadoc) {
    classifier = 'javadoc'
    from javadoc.destinationDir
}

artifacts {
    // archives javadocJar
    archives sourcesJar
}
project.afterEvaluate {
    publishing {
        repositories {
            maven { 
                url "${System.env.HOME}/.m2/repository/"
            }
        }
        publications {
           MyLib(MavenPublication) {
                groupId publishedGroupId
                artifactId artifact
                version libraryVersion
                //from components.android
                artifact "$buildDir/outputs/aar/myLibModule-debug.aar"
                // from components.java
                artifact sourcesJar
                pom.withXml {
                    // Name
                    asNode().appendNode('name', "myLibModule")
                    // Description
                    asNode().appendNode('description', 'A wrapper library for basic Android')
                    // Dependencies
                    def dependenciesNode = asNode().getAt('dependencies')[0]
                    if (dependenciesNode == null) {
                        dependenciesNode = asNode().appendNode('dependencies')
                    }
                    // Add all that are 'compile'
                    configurations.api.allDependencies.each {
                        def dependencyNode = dependenciesNode.appendNode('dependency')
                        dependencyNode.appendNode('groupId', it.group)
                        dependencyNode.appendNode('artifactId', it.name)
                        dependencyNode.appendNode('version', it.version)
                    }
                }
            }
        }
    }
}


project.ext {
    mavDevelopers = ["en":"Naval"]
    mavSiteUrl = "giturlwithout_dot_git"
    mavGitUrl = mavSiteUrl + '.git'
    mavProjectName = 'myLibModule'
    mavPublishToInternalRepo = true
    mavRepoInternalUrl =  "${System.env.HOME}/.m2/repository/"
    mavLibraryLicenses = ["Apache-2.0":'http://www.apache.org/licenses/LICENSE-2.0.txt']
    mavLibraryDescription = "description"
    mavPluginVersion = '1.0.3' //check on raw.githubusercontent.com
}
apply from: "https://raw.githubusercontent.com/sky-uk/gradle-maven-plugin/${project.mavPluginVersion}/gradle-mavenizer.gradle"


this how i build.gradle look like

**Comment by Fed93** [comment]
@NavalKishor if you use the gradle maven plugin you have to remove all the code relative to `dcendents` (that is also **deprecated**).

Please try removig these portions of code:
```
plugins {
// This plugin allows publishing android libraries
    id "com.github.dcendents.android-maven" version "2.1"
}
```

```
apply plugin: 'com.github.dcendents.android-maven'
```

```
apply plugin: 'maven-publish'
```

```
task sourcesJar(type: Jar) {
    classifier = 'sources'
    from android.sourceSets.main.java.srcDirs
}

task javadoc(type: Javadoc) {
    source = android.sourceSets.main.java.srcDirs
    classpath += configurations.compile
    classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
    failOnError = false
}

task javadocJar(type: Jar, dependsOn: javadoc) {
    classifier = 'javadoc'
    from javadoc.destinationDir
}

artifacts {
// archives javadocJar
    archives sourcesJar
}
project.afterEvaluate {
    publishing {
        repositories {
            maven {
                url "${System.env.HOME}/.m2/repository/"
            }
        }
        publications {
            MyLib(MavenPublication) {
                groupId publishedGroupId
                artifactId artifact
                version libraryVersion
//from components.android
                artifact "$buildDir/outputs/aar/myLibModule-debug.aar"
// from components.java
                artifact sourcesJar
                pom.withXml {
// Name
                    asNode().appendNode('name', "myLibModule")
// Description
                    asNode().appendNode('description', 'A wrapper library for basic Android')
// Dependencies
                    def dependenciesNode = asNode().getAt('dependencies')[0]
                    if (dependenciesNode == null) {
                        dependenciesNode = asNode().appendNode('dependencies')
                    }
// Add all that are 'compile'
                    configurations.api.allDependencies.each {
                        def dependencyNode = dependenciesNode.appendNode('dependency')
                        dependencyNode.appendNode('groupId', it.group)
                        dependencyNode.appendNode('artifactId', it.name)
                        dependencyNode.appendNode('version', it.version)
                    }
                }
            }
        }
    }
}
```

If you want you can check this repo to see an example of how to use gradle maven plugin: https://github.com/gotev/localiser/tree/master/example/generated/android

Also if you want to publish your library in .m2/repository you can set `mavPublishToMavenLocal` to true in your configuration instead of 
```
mavPublishToInternalRepo = true
mavRepoInternalUrl = "${System.env.HOME}/.m2/repository/"
```

**Comment by NavalKishor** [comment]
I have removed the above code and followed the example as well but same error

**Comment by Fed93** [comment]
@NavalKishor i noticed that in your build.gradle file is missing the `release` buildType.
Please re-add `release` type after `debug` once.
Like this:

```
buildTypes {
        debug {
            testCoverageEnabled true
            minifyEnabled false
            debuggable true
            jniDebuggable true
            renderscriptDebuggable true
            zipAlignEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }
```

**Comment by NavalKishor** [comment]
Hi @Fed93 added but yet build is failing with same error
i have added below line
release {
            minifyEnabled true
            debuggable false
            jniDebuggable false
            renderscriptDebuggable false
            zipAlignEnabled true
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }

**Comment by NavalKishor** [comment]
I have looked into the gradle window->myLibModule->Tasks->other->
bundleTestDebugAar
bundleTestDebugAndroidTestResources
bundleTestReleaseAar

Like that for other flavours but not 'bundleReleaseAar'
that mean this 'bundleReleaseAar' should be "bundle*ReleaseAar"
or something which can iterate for all flavour one by one, 
or pick based on the selected variant automatically.  

Also, I would like to bring to your notice, if we look into the Gradle window of app 
gradle window->app->
build->bundle
build->bundleDebug
build->bundleRelease
Tasks->other-> same as myLibModule and other without Aar.
so when we will select our build variant respective Gradle task will get executed.

**Comment by skymontif** [comment]
@NavalKishor thanks for your feedback, I make checks on this and I'll fix the problem.

**Comment by Fed93** [comment]
@NavalKishor thanks for your feedback, I make checks on this and I'll fix the problem.

**Comment by vincent-paing** [comment]
Has this been fixed? seem to be still existing

**Comment by gotev** [comment]
fix released in `1.1.0`
